internal/controllers/token_exchange.go

```diff
@@ -155,3 +218,152 @@ func (t *TokenExchangeController) GetDeviceCommandPermissionWithScope(c *fiber.C
 		Token: tk,
 	})
 }
+
+func (t *TokenExchangeController) fetchFromIPFS(cid string) (string, error) {
+	// Remove the "ipfs://" prefix if it exists
+	// TODO Mayber a Regex is better. We should remove "ipfs://" from the SACD source
+	cid = strings.TrimPrefix(cid, "ipfs://")
+
+	url := fmt.Sprintf("https://assets.dimo.xyz//ipfs/%s", cid)
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", fmt.Errorf("failed to fetch from IPFS: %w", err)
+	}
+	defer resp.Body.Close()
```

Comment:
Couple of suggestions here
1. Make the Base URL configurable by adding it to settings then putting it on the controller struct in New(...)
2. Use [url.JoinPath](https://pkg.go.dev/net/url#JoinPath) Not only does it ensure you create a valid URL, But it also does character escaping so we are not vulnerable to XSS attacks when dealing with user provided data.
3. Create the request first with a context. If we do no then this request can hang indefinitely 
```suggestion
	ipfsURL := url.JoinPath(t.IPFSBaseURL, cid)
	req, err := http.NewRequestWithContext(ctx, http.MethodGet, ipfsURL, nil)
	if err != nil {
		return "", fmt.Errorf("failed to create HTTP request: %w", err)
	}
	resp, err := http.DefaultClient.Do(req)
	if err != nil {
		return "", fmt.Errorf("failed to execute request: %w", err)
	}
	defer resp.Body.Close()
```